tests/ui/mir/mir_let_chains_drop_order.rs

```diff
@@ -73,21 +85,26 @@ fn main() {
                         None
                     } else {
                         Some(true)
-                    }
-                ).extra
+                    },
+                )
+                .extra,
             ),
             d(15, None),
             &d(16, None),
             d(17, None),
-            if let DropLogger { .. } = d(18, None) && let DropLogger { .. } = d(19, None) {
+            if let DropLogger { .. } = d(18, None)
+                && let DropLogger { .. } = d(19, None)
+            {
                 d(20, None)
-            }
-            else {
+            } else {
                 // 10 is not constructed
                 d(21, None)
             },
-            panic::panic_any(InjectedFailure)
+            panic::panic_any(InjectedFailure),
         );
     });
+    #[cfg(not(edition2024))]
     assert_eq!(get(), vec![20, 17, 15, 11, 19, 18, 16, 12, 14, 13]);
+    #[cfg(edition2024)]
+    assert_eq!(get(), vec![14, 13, 19, 18, 20, 17, 15, 11, 16, 12]);
```

Comment:
Question [REV6]: ditto here, cfg'd out code does not seem like what we want.

Reply:
The purpose of this test is to pair up the observed drop order from the program print-out with the built MIR. While we have borrow checker tests to cover new lifetime violation due to this edition change, unfortunately it is a category of semantic changes that is only observable in runtime, which is captured by this test. Probably a UI test error annotation is unsuitable.